## Process Analysis
Taking a look at processes first.

In [1]:
python3 ../../volatility3/vol.py -f ../S039/Memory/ACC-01-dc8b5b3e.vmem windows.pstree.PsTree

Volatility 3 Framework 2.0.0

PID	PPID	ImageFileName	Offset(V)	Threads	Handles	SessionId	Wow64	CreateTime	ExitTime

4	0	System	0xfa8003c78b30	87	543	N/A	False	2021-06-23 21:14:35.000000 	N/A
* 240	4	smss.exe	0xfa8004e99b30	2	31	N/A	False	2021-06-23 21:14:35.000000 	N/A
320	312	csrss.exe	0xfa8004f85b30	9	591	0	False	2021-06-23 21:14:36.000000 	N/A
* 1368	320	conhost.exe	0xfa8005fdc060	2	32	0	False	2021-06-23 21:14:39.000000 	N/A
* 1540	320	conhost.exe	0xfa8006b6a060	2	34	0	False	2021-06-23 21:14:39.000000 	N/A
372	312	wininit.exe	0xfa8005a536c0	3	76	0	False	2021-06-23 21:14:36.000000 	N/A
* 480	372	services.exe	0xfa8005b244b0	7	235	0	False	2021-06-23 21:14:36.000000 	N/A
** 2436	480	msdtc.exe	0xfa8004efb800	12	146	0	False	2021-06-23 21:14:45.000000 	N/A
** 1400	480	sysmon.exe	0xfa8006b35b30	11	306	0	False	2021-06-23 21:14:39.000000 	N/A
** 1036	480	spoolsv.exe	0xfa8005f4fb30	12	277	0	False	2021-06-23 21:14:38.000000 	N/A
** 3724	480	svchost.exe	0xfa8005844060	11	141	0	False	2021-06-23 2

Immediately, `svchost.exe` with PID 3040 is suspicious, as it ran fram `explorer.exe` and not from `services.exe` as most do. Let's take a look at its handles.

In [1]:
python3 ../../volatility3/vol.py -f ../S039/Memory/ACC-01-dc8b5b3e.vmem windows.handles.Handles --pid 3040

The below script needs to be able to find the current output cell; this is an easy method to get it.

Volatility 3 Framework 2.0.0

PID	Process	Offset	HandleValue	Type	GrantedAccess	Name

3040	svchost.exe	0xf8a0021d1900	0x4	Key	0x9	MACHINE\SOFTWARE\MICROSOFT\WINDOWS NT\CURRENTVERSION\IMAGE FILE EXECUTION OPTIONS
3040	svchost.exe	0xf8a0005a8740	0x8	Directory	0x3	KnownDlls
3040	svchost.exe	0xfa8004db87b0	0xc	File	0x100020	\Device\HarddiskVolume1\Windows\System32
3040	svchost.exe	0xfa8005d585f0	0x10	Event	0x1f0003	
3040	svchost.exe	0xf8a007986060	0x14	Key	0x20019	MACHINE\SYSTEM\CONTROLSET001\CONTROL\NLS\SORTING\VERSIONS
3040	svchost.exe	0xfa8004a97640	0x18	ALPC Port	0x1f0001	
3040	svchost.exe	0xfa8005f55470	0x1c	ALPC Port	0x1f0001	
3040	svchost.exe	0xfa8006b1b3e0	0x20	Mutant	0x1f0001	
3040	svchost.exe	0xf8a000eab680	0x24	Key	0x1	MACHINE\SYSTEM\CONTROLSET001\CONTROL\SESSION MANAGER
3040	svchost.exe	0xfa8005d3b370	0x28	Mutant	0x1f0001	
3040	svchost.exe	0xf8a001ba0880	0x2c	Key	0x20019	MACHINE
3040	svchost.exe	0xfa8005d45d40	0x30	Event	0x1f0003	
3040	svchost.exe	0xfa8005d41210	0x34	Mutant	0x1

3040	svchost.exe	0xfa8005d53680	0x20c	EtwRegistration	0x804	
3040	svchost.exe	0xfa8005d535c0	0x210	EtwRegistration	0x804	
3040	svchost.exe	0xf8a000575340	0x214	Key	0xf003f	USER\S-1-5-21-1071627472-1312190459-2730622649-1660_CLASSES
3040	svchost.exe	0xf8a00139fcf0	0x218	Key	0x1	USER\S-1-5-21-1071627472-1312190459-2730622649-1660\SOFTWARE\MICROSOFT\WINDOWS\CURRENTVERSION\EXPLORER
3040	svchost.exe	0xf8a000f2ab00	0x21c	Section	0x4	__ComCatalogCache__
3040	svchost.exe	0xf8a00220ae50	0x220	Key	0x8	USER\S-1-5-21-1071627472-1312190459-2730622649-1660\SOFTWARE\MICROSOFT\WINDOWS NT\CURRENTVERSION
3040	svchost.exe	0xf8a00111acc0	0x224	Key	0x8	MACHINE\SOFTWARE\MICROSOFT\WINDOWS NT\CURRENTVERSION\APPCOMPATFLAGS
3040	svchost.exe	0xf8a001e92a20	0x228	Key	0xf003f	USER\S-1-5-21-1071627472-1312190459-2730622649-1660_CLASSES
3040	svchost.exe	0xfa8003df9550	0x22c	Event	0x100001	MaximumCommitCondition
3040	svchost.exe	0xf8a000f2ab00	0x230	Section	0x4	__ComCatalogCache__
3040	svchost.exe	0xfa8006b14540	0x23

3040	svchost.exe	0xfa8005ae5f10	0x36c	Semaphore	0x100003	
3040	svchost.exe	0xfa8005b1bf10	0x370	Semaphore	0x100003	
3040	svchost.exe	0xfa8004e90ad0	0x374	Semaphore	0x100003	
3040	svchost.exe	0xfa8004e91830	0x378	Semaphore	0x100003	
3040	svchost.exe	0xfa8004d509b0	0x37c	Semaphore	0x100003	
3040	svchost.exe	0xfa8005b2bb00	0x380	Semaphore	0x100003	
3040	svchost.exe	0xfa8005c10e80	0x384	Event	0x1f0003	
3040	svchost.exe	0xf8a002279ad0	0x388	Key	0x20019	MACHINE\SYSTEM\CONTROLSET001\SERVICES\WINSOCK2\PARAMETERS\PROTOCOL_CATALOG9
3040	svchost.exe	0xfa80057d5560	0x38c	Event	0x1f0003	
3040	svchost.exe	0xf8a0078a3e20	0x390	Key	0x20019	MACHINE\SYSTEM\CONTROLSET001\SERVICES\WINSOCK2\PARAMETERS\NAMESPACE_CATALOG5
3040	svchost.exe	0xfa8005d2d070	0x394	EtwRegistration	0x804	
3040	svchost.exe	0xfa8005e4ae70	0x398	EtwRegistration	0x804	
3040	svchost.exe	0xfa8004dc62a0	0x39c	Event	0x1f0003	
3040	svchost.exe	0xfa8004e1db70	0x3a0	Mutant	0x1f0001	WininetConnectionMutex
3040	svchost.exe	0xfa8005d9c080	0x3a4	

Nothing jumps out as suspicious here. Let's take a look at DLLs.

In [2]:
python3 ../../volatility3/vol.py -f ../S039/Memory/ACC-01-dc8b5b3e.vmem windows.dlllist.DllList --pid 3040

Volatility 3 Framework 2.0.0

PID	Process	Base	Size	Name	Path	LoadTime	File output

3040	svchost.exe	0x11a0000	0x8000	svchost.exe	C:\Users\Chad.Chan\AppData\Roaming\Microsoft\Windows\Start Menu\Programs\Startup\svchost.exe	N/A	Disabled
3040	svchost.exe	0x76f30000	0x1a9000	ntdll.dll	C:\Windows\SYSTEM32\ntdll.dll	N/A	Disabled
3040	svchost.exe	0x7fef4350000	0x6f000	MSCOREE.DLL	C:\Windows\SYSTEM32\MSCOREE.DLL	2021-06-23 21:14:56.000000 	Disabled
3040	svchost.exe	0x76d10000	0x11f000	KERNEL32.dll	C:\Windows\system32\KERNEL32.dll	2021-06-23 21:14:56.000000 	Disabled
3040	svchost.exe	0x7fefd1e0000	0x6b000	KERNELBASE.dll	C:\Windows\system32\KERNELBASE.dll	2021-06-23 21:14:56.000000 	Disabled
3040	svchost.exe	0x7fefd600000	0xdb000	ADVAPI32.dll	C:\Windows\system32\ADVAPI32.dll	2021-06-23 21:14:56.000000 	Disabled
3040	svchost.exe	0x7fefef60000	0x9f000	msvcrt.dll	C:\Windows\system32\msvcrt.dll	2021-06-23 21:14:56.000000 	Disabled
3040	svchost.exe	0x7fefdfd0000	0x1f000	sechost.dll	C:\Windows\SYSTEM

3040	svchost.exe	0x7fefc0a0000	0x7000	wshtcpip.dll	C:\Windows\System32\wshtcpip.dll	2021-06-23 21:15:00.000000 	Disabled
3040	svchost.exe	0x7fefc6a0000	0x7000	wship6.dll	C:\Windows\System32\wship6.dll	2021-06-23 21:15:00.000000 	Disabled
3040	svchost.exe	0x7fefa7c0000	0x53000	fwpuclnt.dll	C:\Windows\System32\fwpuclnt.dll	2021-06-23 21:15:00.000000 	Disabled
3040	svchost.exe	0x7fefbfd0000	0xc000	VERSION.dll	C:\Windows\system32\VERSION.dll	2021-06-23 21:15:00.000000 	Disabled

Progress:    0.00		Scanning VmwareLayer using PageMapScanner

Progress:    0.00		Scanning VmwareLayer using PageMapScanner

Progress:   23.33		Scanning VmwareLayer using PageMapScanner

Progress:    0.00		Scanning VmwareLayer using PageMapScanner

Progress:  100.00		Stacking attempts finished               

Progress:    0.00		Scanning memory_layer using BytesScanner 

Progress:    0.31		Scanning memory_layer using BytesScanner 

Progress:    0.63		Scanning memory_layer using BytesScanner 

Progress:    0.00		Scann

Nothing jumps out as immediately odd within those DLLs.

After taking a look at ACC-03, it seems `svchost.exe` was copied to the startup folder on this machine. Since taking a look at its DLLs and handles didn't seem to yield anything, it might help to dump the program and floss it.

In [6]:
python3 ../../volatility3/vol.py -q -o ../svchost_dump -f ../S039/Memory/ACC-01-dc8b5b3e.vmem windows.pslist.PsList --pid 3040 --dump

Volatility 3 Framework 2.0.0

PID	PPID	ImageFileName	Offset(V)	Threads	Handles	SessionId	Wow64	CreateTime	ExitTime	File output

3040	2812	svchost.exe	0xfa8005fc2b30	7	301	1	False	2021-06-23 21:14:56.000000 	N/A	pid.3040.0x11a0000.dmp


Let's floss this program to get some insight into what's really going on.

In [7]:
../../floss ../svchost_dump/pid.3040.0x11a0000.dmp

ERROR:floss:Vivisect failed to load the input file: float division by zero
FLOSS static ASCII strings
!This program cannot be run in DOS mode.
.text
`.rsrc
@.reloc
AQAPRQVH1
JJM1
RAQH
AXAX^YZAXAYAZH
XAYZH
wininet
APAPA
AQAQj
/image-directory/dhl.jpg
9tpY
f0rU
Host: backup-helper.com
Connection: close
Accept-Encoding: br
User-Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/42.0.2311.135 Safari/537.36 Edge/12.246
VR-o
/RHah
XXXH
backup-helper.com
BSJB
v4.0.30319
#Strings
#GUID
#Blob
<Module>
rk-2.exe
Program
HashPrototype
mscorlib
System
Object
MulticastDelegate
SW_HIDE
VirtualAlloc
VirtualFree
GetConsoleWindow
ShowWindow
Main
AES_Encrypt
changePerm
.ctor
Invoke
IAsyncResult
AsyncCallback
BeginInvoke
EndInvoke
lpAddress
dwSize
flAllocationType
flProtect
dwFreeType
hWnd
nCmdShow
bytesToBeEncrypted
passwordBytes
oldFilePath
object
method
InputText
callback
result
System.Runtime.CompilerServices
CompilationRelaxationsAttribute
RuntimeCompatibil

# Network Analysis

In [3]:
python3 ../../volatility3/vol.py -f ../S039/Memory/ACC-01-dc8b5b3e.vmem windows.netstat.NetStat

Volatility 3 Framework 2.0.0

Offset	Proto	LocalAddr	LocalPort	ForeignAddr	ForeignPort	State	PID	Owner	Created

0xfa8006e676e0	TCPv4	127.0.0.1	49190	127.0.0.1	49191	ESTABLISHED	1532	python.exe	N/A
0xfa8004d43b40	TCPv4	172.17.11.170	49299	18.213.72.214	443	ESTABLISHED	2792	winlogbeat.exe	N/A
0xfa800449bba0	TCPv4	172.17.11.170	49633	23.47.50.93	443	ESTABLISHED	1320	iexplore.exe	N/A
0xfa8006e08b50	TCPv4	172.17.11.170	49295	10.20.1.11	8000	ESTABLISHED	4020	Velociraptor.e	N/A
0xfa80042107c0	TCPv4	172.17.11.170	49591	52.216.9.251	443	CLOSE_WAIT	2960	user.exe	N/A
0xfa800433acf0	TCPv4	172.17.11.170	49640	13.225.51.5	443	ESTABLISHED	1320	iexplore.exe	N/A
0xfa8005e13010	TCPv4	172.17.11.170	49298	3.225.196.73	443	ESTABLISHED	2792	winlogbeat.exe	N/A
0xfa8004473360	TCPv4	172.17.11.170	49627	13.225.51.5	443	ESTABLISHED	1320	iexplore.exe	N/A
0xfa8004abfba0	TCPv4	127.0.0.1	49197	127.0.0.1	49196	ESTABLISHED	1532	python.exe	N/A
0xfa8006fcc9d0	TCPv4	127.0.0.1	4510	127.0.0.1	49184	ESTABLISHED	1532	python.

In [4]:
python3 ../../volatility3/vol.py -f ../S039/Memory/ACC-01-dc8b5b3e.vmem windows.netscan.NetScan

Volatility 3 Framework 2.0.0

Offset	Proto	LocalAddr	LocalPort	ForeignAddr	ForeignPort	State	PID	Owner	Created

0x7553ec0	UDPv4	172.17.11.170	52180	*	0		3724	svchost.exe	2021-06-23 21:16:41.000000 
0x13cc05d70	UDPv4	127.0.0.1	64067	*	0		876	svchost.exe	2021-06-23 21:14:46.000000 
0x13cc5bcf0	TCPv6	-	0	6830:8005:80fa:ffff:6830:8005:80fa:ffff	0	CLOSED	488	lsass.exe	N/A
0x13cc70ad0	UDPv4	0.0.0.0	123	*	0		960	svchost.exe	2021-06-23 21:14:42.000000 
0x13cc70d70	UDPv4	0.0.0.0	123	*	0		960	svchost.exe	2021-06-23 21:14:42.000000 
0x13cc70d70	UDPv6	::	123	*	0		960	svchost.exe	2021-06-23 21:14:42.000000 
0x13ce08b50	TCPv4	172.17.11.170	49295	10.20.1.11	8000	ESTABLISHED	4020	Velociraptor.e	N/A
0x13ce333d0	TCPv4	0.0.0.0	49211	0.0.0.0	0	LISTENING	488	lsass.exe	-
0x13ce40340	TCPv4	127.0.0.1	49193	127.0.0.1	49192	ESTABLISHED	1532	python.exe	N/A
0x13ce676e0	TCPv4	127.0.0.1	49190	127.0.0.1	49191	ESTABLISHED	1532	python.exe	N/A
0x13ce69240	TCPv4	-	0	56.11.230.5	0	CLOSED	488	lsass.exe	N/A
0x13ce71a60	TCP

0x13f8b7ec0	UDPv4	127.0.0.1	51633	*	0		2172	OUTLOOK.EXE	2021-06-23 21:56:42.000000 
0x13f8d3cf0	TCPv4	172.17.11.170	49641	13.225.51.5	443	ESTABLISHED	1320	iexplore.exe	N/A
0x13f9852d0	TCPv4	172.17.11.170	49629	23.40.185.117	443	ESTABLISHED	1320	iexplore.exe	N/A
0x13fa107c0	TCPv4	172.17.11.170	49591	52.216.9.251	443	CLOSE_WAIT	2960	user.exe	N/A
0x13fa11400	TCPv4	172.17.11.170	49389	10.20.1.11	8000	ESTABLISHED	4020	Velociraptor.e	N/A
0x13fa1b160	UDPv4	127.0.0.1	62618	*	0		2104	iexplore.exe	2021-06-23 21:35:45.000000 
0x13fa68600	TCPv4	172.17.11.170	49637	23.47.50.75	443	ESTABLISHED	1320	iexplore.exe	N/A
0x13fa6c3c0	TCPv4	172.17.11.170	49632	23.47.50.93	443	ESTABLISHED	1320	iexplore.exe	N/A
0x13fa6d9c0	TCPv4	-	0	-	0	CLOSED	1320	iexplore.exe	-
0x13fa8d980	TCPv4	172.17.11.170	49643	162.247.242.19	443	ESTABLISHED	1320	iexplore.exe	N/A
0x13faa81f0	TCPv4	172.17.11.170	49636	23.47.50.93	443	ESTABLISHED	1320	iexplore.exe	N/A
0x13fb23490	TCPv4	172.17.11.170	49635	23.47.50.93	443	ESTABLISHED	1320	

There's doesn't seem to be anything malicious taking place here. Most connections are local, a few are Akamai Technologies (cloud service provider). The one odd process is an `lsass.exe` process with PID 488 connecting to 56.11.230.5 (USPS).